src/ucs/config/parser.c

```diff
@@ -379,6 +398,111 @@ int ucs_config_sprintf_time(char *buf, size_t max, void *src, const void *arg)
     return 1;
 }
 
+int ucs_config_sscanf_bw(const char *buf, void *dest, const void *arg)
+{
+    double *dst     = (double*)dest;
+    char    str[16] = {0};
+    int     offset  = 0;
+    size_t  divider;
+    double  units;
+    double  value;
+    int     num_fields;
+
+    num_fields = sscanf(buf, "%lf%16s", &value, str);
+    if (num_fields < 2) {
+        return 0; /* bytes or bits? */
+    }
+
+    ucs_assert(num_fields == 2);
+
+    units = ucs_config_parser_quantity_prefix_value(str[0]);
+
+    if (!units) {
+        return 0;
+    }
+
+    offset = (units == 1.0) ? 0 : 1;
+
+    if (str[offset] == 'B') {
+        divider = 1;
+    } else if (str[offset] == 'b') {
+        divider = 8;
+    } else {
+        return 0;
+    }
+
+    offset++;
+
+    if (strcmp(str + offset, "ps") &&
+        strcmp(str + offset, "/s") &&
+        strcmp(str + offset, "s")) {
+        return 0;
+    }
+
+    ucs_assert((divider == 1) || (divider == 8)); /* bytes or bits */
+    *dst = value * units / divider;
+    return 1;
+}
+
+int ucs_config_sprintf_bw(char *buf, size_t max, void *src, const void *arg)
+{
+    double value = *(double*)src;
+    size_t len;
+
+    ucs_memunits_to_str((size_t)value, buf, max);
```

Comment:
looks like value can be truncated here..

Reply:
right, but it should be enough accurate to print

Reply:
ok